code/contracts/ONEWallet.sol

```diff
@@ -206,12 +214,12 @@ contract ONEWallet {
         (uint32 ct, bool completed) = _findCommit(hash);
         require(ct > 0, "Cannot find commit for this transaction");
         uint32 counter = ct / interval - t0;
-        require(counter == index, "Provided index does not match commit timestamp");
+        require(counter == index, "Provided index does not match committed timestamp");
         uint8 expectedNonce = nonces[counter];
-        require(nonce >= expectedNonce, "Nonce is too low");
+        require(nonce >= expectedNonce, "Nonce is too low"); // IH: should not be here equality? One commit object is associated with the one nonce (it is part of commited hash)
         require(!completed, "Commit is already completed");
         // this should not happen (since old commit should be cleaned up already)
-        require(_isRevealTimely(ct), "Reveal is too late. Please re-commit");
+        require(_isRevealTimely(ct), "Reveal is too late. Please re-commit"); // IH: What might happen if we'd increase REVEAL_MAX_DELAY 10x or 100x? I do not see any advantage of keeping it short since MITM attacker gets only extended 32B OTP while attacker that tampers with the client can do anything regardless of this lenght. This brings me again to using HOTPs.
```

Comment:
That would make the storage for commit very large when an attacker commits a lot of garbage. It is not much of a problem if it is kept temporary, since the attacker paid for it. But it must not be made permanent (or stays too long, in the case of making REVEAL_MAX_DELAY 10x or 100x), otherwise the size of the contract grows unbounded without producing additional value. In practice, the reveal should happen immediately after a commit. It is not reasonable for a client to wait for too long after committing an operation aside from bad network conditions.

Reply:
It's at the attacker's expense anyway. Therefore, the attacker is not incentivized to do it since he pays for it. And Harmony blockchain wants users who pays for txs. Hence, to me, growing blockchain is an unrelated issue - such attacker will not occur for sure (unless something is improperly designed - however, this we can influence).

Reply:
On a second thought, perhaps we should allow an "untimely" reveal. There are legitimate use cases for this in dApps: sign a transaction at first, execute it later. The smart contract wallet equivalent would be to commit (with signature) at first, reveal later.